File: py/gateway/tls_audit.py
#!/usr/bin/env python3
"""Verify the TLS identity of the configured NEAR AI Cloud Gateway.

The peer certificate is observed while the Gateway attestation is fetched. A
later TLS connection could reach a different backend, so this utility does not
compare an unrelated follow-up connection or a ``tls_certificate`` copy.
"""

from __future__ import annotations

import asyncio

from py.gateway.cloud_api import (
    API_KEY,
    BASE_URL,
)
from py.gateway.deployment_audit import audit_gateway_attestation


async def main() -> None:
    """Fetch and verify Gateway evidence plus its observed TLS peer."""

    if not API_KEY:
        raise RuntimeError("API_KEY is required to fetch Cloud API Gateway evidence")

    print("========================================")
    print("🔐 Gateway TLS attestation")
    print("========================================")
    print("Gateway endpoint:", BASE_URL)

    await audit_gateway_attestation()


if __name__ == "__main__":
    asyncio.run(main())

File: ts/gateway/tls_audit.ts
#!/usr/bin/env node
/**
 * Verify the TLS identity of the configured NEAR AI Cloud Gateway.
 *
 * The peer certificate is observed while the Gateway attestation is fetched.
 * A later TLS connection could reach a different backend, so this script does
 * not compare an unrelated follow-up connection or a tls_certificate copy.
 */

import { auditGatewayAttestation } from './deployment_audit';

function cloudApiBaseUrl(): string {
  return process.env.BASE_URL || 'https://cloud-api.near.ai';
}

async function main(): Promise<void> {
  if (!process.env.API_KEY) {
    throw new Error('API_KEY is required to fetch Cloud API Gateway evidence');
  }

  console.log('========================================');
  console.log('🔐 Gateway TLS attestation');
  console.log('========================================');
  console.log('Gateway endpoint:', cloudApiBaseUrl());

  await auditGatewayAttestation();
}

if (require.main === module) {
  main().catch((error) => {
    console.error(
      '\nVerification failed:',
      error instanceof Error ? error.message : error,
    );
    process.exitCode = 1;
  });
}
